### PR TITLE
feat(query): added PostgreSQL Server Infrastructure Encryption Disabled for Te…

### PR DESCRIPTION
**Pull request by tiagocruz20**
…rraform

**Proposed Changes**
- added a query to check if infrastructure encryption is enabled in PostgreSQL Server

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption/metadata.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "id": "6425c98b-ca4e-41fe-896a-c78772c131f8",
+  "queryName": "PostgreSQL Server Infrastructure Encryption",
```
Comment (rafaela-soares) [inline assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption/metadata.json:3]: Maybe we can rename it for something more intuitive like `"queryName": "PostgreSQL Server Infrastructure Encryption Disabled"`

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption/query.rego [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	pg := input.document[i].resource.azurerm_postgresql_server[name]
+
+	pg.infrastructure_encryption_enabled == false
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("azurerm_postgresql_server[%s].infrastructure_encryption_enabled", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is set to true", [name]),
+		"keyActualValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is not set to true", [name]),
```
Comment (felipe-avelar) [inline assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption/query.rego]:
```suggestion
		"keyActualValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is false", [name]),
```

File: assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption_disabled/query.rego [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	pg := input.document[i].resource.azurerm_postgresql_server[name]
+
+	pg.infrastructure_encryption_enabled == false
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("azurerm_postgresql_server[%s].infrastructure_encryption_enabled", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is set to true", [name]),
+		"keyActualValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is not set to true", [name]),
```
Comment (felipe-avelar) [inline assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption_disabled/query.rego:30]:
```suggestion
		"keyActualValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is set to false", [name]),
```

File: assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption_disabled/query.rego [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+CxPolicy[result] {
+	pg := input.document[i].resource.azurerm_postgresql_server[name]
+
+	not common_lib.valid_key(pg, "infrastructure_encryption_enabled")
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("azurerm_postgresql_server[%s]", [name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is defined and not null", [name]),
```
Comment (felipe-avelar) [inline assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption_disabled/query.rego:14]:
```suggestion
		"keyExpectedValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is defined and set to true", [name]),
```

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

File: assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption_disabled/query.rego [hunk 5]
```diff
@@ -11,8 +11,8 @@ CxPolicy[result] {
 		"documentId": input.document[i].id,
 		"searchKey": sprintf("azurerm_postgresql_server[%s]", [name]),
 		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is defined and not null", [name]),
-		"keyActualValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is undefined or null", [name]),
+		"keyExpectedValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is defined and set to true", [name]),
+		"keyActualValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is undefined or set to false", [name]),
```
Comment (rafaela-soares) [inline assets/queries/terraform/azure/postgre_sql_server_infrastructure_encryption_disabled/query.rego:15]: I think we should maintain `"keyActualValue": sprintf("'azurerm_postgresql_server[%s].infrastructure_encryption_enabled' is undefined or null", [name]),`

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]

**Pull request merged**